gaek/ndb_json.py

```diff
@@ -120,9 +133,21 @@ def encode_basevalue(obj):
 }
 
 
+ALTERNATIVE_NDB_TYPE_ENCODING = {
+  ndb.Key: encode_key_as_pair,
+}
+
+
 class NdbEncoder(json.JSONEncoder):
   """Extend the JSON encoder to add support for NDB Models."""
 
+  # Whether to fetch entities specified by ndb.Key or to return them as pairs.
+  _fetch_models=True
+
+  def __init__(self, **kwargs):
+    self._fetch_models = kwargs.pop('fetch_models', True)
```

Comment:
Let's discuss some options for how we can allow callers to define how `ndb.Keys` are serialized.

1. I think I makes sense to prefix all `ndb_json` arguments with `ndb_` (i.e. `ndb_fetch_models=True`.
2. I personally prefer boolean arguments over string values for flags, even if it does lead to some complexity in this function for validation.
3. To maintain backwards compatibility, the default should be to encode keys as Entities.
4. It seems like the following options make sense to me:
  * `ndb_keys_as_entities`
  * `ndb_keys_as_pairs`
  * `ndb_keys_as_urlsafe`
5. If more than one of those options are set to `True`, we'll need to throw an exception to help developers.
6. To help with performance, it might make sense to copy the `NDB_TYPE_ENCODING` dictionary when this class is instantiated so that the `ndb.Key` value can be switched to the appropriate function. That should remove the need for any special-case conditionals during the serialization process below.

Thoughts?